### PR TITLE
GitHub actions v1

### PR DESCRIPTION
**Pull request by tristan0x**
This contribution adds the first brick of CI of this project by checking:
* the formatting of both CMake and Python code
* the static analysis (with flake8) of the Python code

---
blocked by #123

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Format Python and CMake code
+9fbb3c4d75ea8c2e8b49e5625e1d5f58f515539f
```
Comment (olupton) [inline .git-blame-ignore-revs:2]: Just a reminder that the PR needs to not get squashed/rebased/... in a way that makes this invalid.

File: .github/workflows/check_cmake_format.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  build:
+    name: check-cmake-format
+    runs-on: ubuntu-20.04
+    # Run on external PRs, but not internal PRs as they'll be run by the push
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    steps:
+        - name: Fetch repository
+          uses: actions/checkout@v1
```
Comment (olupton) [inline .github/workflows/check_cmake_format.yml:13]: Any particular reason for pinning an old version? IIUC the latest is v3.

**Review by olupton — COMMENTED** [review]
Looks reasonable to me; I had a couple of comments and I don't think `.ci/.check_cmake_format.sh.swp` is meant to be committed?

File: .git-blame-ignore-revs (hunk 1 above)
Comment (tristan0x) [reply .git-blame-ignore-revs:2]: Thanks, indeed I was planning a merge request without squashing

File: .github/workflows/check_cmake_format.yml (hunk 2 above)
Comment (tristan0x) [reply .github/workflows/check_cmake_format.yml]: updated. thanks

**Pull request merged**